contracts/logic/StakingPoolLogicV2.sol

```diff
@@ -42,40 +41,54 @@ library StakingPoolLogicV2 {
     if (poolData.userIndex[user] == 0) {
       return 0;
     }
-    uint256 indexDiff = getRewardIndex(poolData) - poolData.userIndex[user];
 
+    uint256 indexDiff = getRewardIndex(poolData) - poolData.userIndex[user];
     uint256 balance = poolData.userPrincipal[user];
-
     uint256 result = poolData.userReward[user] + (balance * indexDiff) / 1e9;
 
     return result;
   }
 
+  function updateRewardPerSecond(StakingPoolV2.PoolData storage poolData, address user) internal {
+    poolData.userReward[user] = getUserReward(poolData, user);
+    poolData.rewardIndex = poolData.userIndex[user] = getRewardIndex(poolData);
```

Comment:
There is no need to update `userReward` and `userIndex` because it does not change any user's status but affects only the pool status.

Reply:
```suggestion
    poolData.rewardIndex = getRewardIndex(poolData);
```